src/rules/dot_product/in1.jl

```diff
@@ -4,5 +4,40 @@ export rule
     x  = mean(m_in2)
     xi = x * weightedmean(m_out)
     w  = correction!(meta, x * precision(m_out) * x')
-    return MvGaussianWeightedMeanPrecision(xi, w)
+    return MvNormalWeightedMeanPrecision(xi, w)
+end
+
+@rule typeof(dot)(:in1, Marginalisation) (m_out::UnivariateNormalDistributionsFamily, m_in2::PointMass, meta::AbstractCorrection) = begin
+    x  = mean(m_in2)
+    xi = x * weightedmean(m_out)
+    w  = correction!(meta, x * precision(m_out) * x')
+    return NormalWeightedMeanPrecision(xi, w)
+end
+
+@rule typeof(dot)(:in1, Marginalisation) (m_out::UnivariateNormalDistributionsFamily, m_in2::PointMass) = begin
+    x  = mean(m_in2)
```

Comment:
Good that you added additional rules, however, in this case it is not desired. The `meta` argument specifies how the precision should be corrected. The `dot` node always has this object (see `dot_product.jl`) which defaults to `TinyCorrection`. So we are safe to assume that it is always available for our rules. Without this correction we might see problems such as numerical stability when performing the calculations. Therefore please remove the rules without the `meta` argument.